pkg/controller/argocd/secret.go

```diff
@@ -379,6 +385,38 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	secret := argoutil.NewSecretWithSuffix(cr.ObjectMeta, "namespaces")
```

Comment:
I think the we should use a suffix of `cluster-in-cluster` instead of `namespaces` here. Because the secret actually defines the connection to the cluster, and the namespace configuration is just a side-effect :)

Reply:
done !